src/transforms/corrcholeskylower.jl

```diff
@@ -0,0 +1,65 @@
+
+####
+#### correlation cholesky Lower
+####
+# Adapted from https://github.com/tpapp/TransformVariables.jl/blob/master/src/special_arrays.jl
+
+const TV = TransformVariables
+using LinearAlgebra
+
+"""
+    CorrCholeskyLower(n)
+Cholesky Lower of a correlation matrix of size `n`.
+Transforms ``n×(n-1)/2`` real numbers to an ``n×n`` lower-triangular matrix `L`, such that
+`L*L'` is a correlation matrix (positive definite, with unit diagonal).
+# Notes
+If
+- `z` is a vector of `n` IID standard normal variates,
+- `σ` is an `n`-element vector of standard deviations,
+- `L` is obtained from `CorrCholeskyLower(n)`,
+then `Diagonal(σ) * L * z` will be a multivariate normal with the given variances and
+correlation matrix `L * L'`.
+"""
+struct CorrCholeskyLower <: TV.VectorTransform
+    n::Int
+    function CorrCholeskyLower(n)
+        # @argcheck n ≥ 1 "Dimension should be positive."
+        new(n)
+    end
+end
+
```

Comment:
I think this can be greatly simplified (and avoid using non-API functions of TransformVariables) by just calling the corresponding functions for `CorrCholeskyUpper`:

```julia
TV.dimension(t::CorrCholeskyLower) = TV.dimension(TV.CorrCholeskyUpper(t.n))

function TV.transform_with(flag::TV.LogJacFlag, t::CorrCholeskyLower, x::AbstractVector, index)
    U, ℓ, index = TV.transform_with(flag, TV.CorrCholeskyUpper(t.n), x, index)
    return U', ℓ, index
end

TV.inverse_eltype(t::CorrCholeskyLower, L::LowerTriangular) = TV.extended_eltype(L)

function TV.inverse_at!(x::AbstractVector, index, t::CorrCholeskyLower, L::LowerTriangular)
    return TV.inverse_at!(x, index, TV.CorrCholeskyUpper(t.n), L')
end
```

Reply:
Great idea!